crates/circuit/src/operations.rs

```diff
@@ -125,6 +125,21 @@ impl Param {
             Param::Obj(ob.clone().unbind())
         })
     }
+
+    /// Clones the [Param] object safely by reference count or copying.
+    pub fn clone_ref(&self, py: Python) -> Self {
+        match self {
+            Param::ParameterExpression(exp) => Param::ParameterExpression(exp.clone_ref(py)),
+            Param::Float(float) => Param::Float(*float),
+            Param::Obj(obj) => Param::Obj(obj.clone_ref(py)),
+        }
+    }
+}
+
+impl AsRef<Param> for Param {
+    fn as_ref(&self) -> &Param {
+        self
+    }
```

Comment:
Might be worth just putting in a comment here about why this exists potentially with a link to the Rust docs bit about why the blanket impl doesn't: https://doc.rust-lang.org/std/convert/trait.AsRef.html#reflexivity